bananas_api/helpers/api_schema.py

```diff
@@ -0,0 +1,186 @@
+    tags = fields.List(fields.String(validate=validate.Length(max=31)))
+
+
+class Author(OrderedSchema):
+    display_name = fields.String(data_key="display-name")
+    openttd = fields.String(allow_none=True)
+    github = fields.String(allow_none=True)
+    developer = fields.String(allow_none=True)
+
+
+class Authors(OrderedSchema):
+    authors = fields.List(fields.Nested(Author))
+
+
+class Dependency(OrderedSchema):
+    content_type = EnumField(ContentType, data_key="content-type", by_value=True)
+    unique_id = fields.String(data_key="unique-id", validate=validate.Length(min=8, max=8))
+    md5sum_partial = fields.String(data_key="md5sum-partial", validate=validate.Length(min=8, max=8))
+
+    @validates_schema
+    def validate_dependency(self, data, **kwargs):
+        if not DEPENDENCY_CHECK:
+            return
+
+        # Check the unique-id exists
+        package = get_indexed_package(data["content_type"], data["unique_id"])
+        if package is None:
+            raise ValidationError(
+                f"Package with unique-id '{data['unique_id']}' does not exist for {data['content_type'].value}."
+            )
+
+        # Check there is any version with that md5sum-partial
+        for version in package["versions"]:
+            if version["md5sum_partial"] == data["md5sum_partial"]:
+                break
+        else:
+            raise ValidationError(
+                f"No version with md5sum-partial '{data['md5sum_partial']}' exist for "
+                f"{data['content_type'].value} with unique-id '{data['unique_id']}'."
+            )
+
+
+class Compatability(OrderedSchema):
+    name = fields.String()
+    conditions = fields.List(fields.String(), validate=validate.Length(min=1, max=2))
+
+    @validates("conditions")
+    def validate_conditions(self, data, **kwargs):
+        if len(data) == 1:
+            if not data[0].startswith((">= ", "< ")):
+                raise ValidationError(
+                    f"Condition can only mark the first client-version this version does or doesn't work for;"
+                    f" expected '>= VERSION' or '< VERSION', got '{data[0]}'."
+                )
+        else:
+            if not data[0].startswith(">= "):
+                raise ValidationError(
+                    f"First condition can only mark the first client-version this version does work for;"
+                    f" expected '>= VERSION', got '{data[0]}'."
+                )
+            if not data[1].startswith("< "):
+                raise ValidationError(
+                    f"Second condition can only mark the first client-version this version doesn't work for;"
+                    f" expected '< VERSION', got '{data[0]}'."
+                )
+
+
+class VersionMinimized(Global):
+    read_only = ["upload_date", "md5sum_partial", "filesize", "license", "availability"]
+    read_only_for_new = ["upload_date", "md5sum_partial", "filesize", "availability"]
```

Comment:
what about "authors"?

Reply:
This comment is too vague for me. What about them? Could you elaborate a bit more what you refer to?

Reply:
Resolved over IRC: authors are part of the Package schema, and currently you cannot edit them (not during upload of a new package, nor while editing an existing) via the API. On upload of a new package, the current user is set as author automatically.

Will add to the wishlist to address this.